### PR TITLE
Unify declaration dependent version

### PR DESCRIPTION
**Pull request by diguage**
The dependencies are declared in many places. 
Unify declaration dependent version so that this is good for upgrading.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -28,20 +28,25 @@ ext {
...
 	aspectjVersion       = "1.9.2"
 	freemarkerVersion    = "2.3.28"
 	groovyVersion        = "2.5.5"
 	hsqldbVersion        = "2.4.1"
 	jackson2Version      = "2.9.8"
+	jaxbApiVersion       = "2.3.1"
 	jettyVersion         = "9.4.14.v20181114"
+	junit4Version        = "4.12"
 	junit5Version        = "5.3.2"
+	jspApiVersion        = "2.3.2-b02"
```
Comment (snicoll) [inline build.gradle]: This should be moved ideally before `junit4Version` to respect the current order.

**Review by snicoll — COMMENTED** [review]
LGTM, I've added a suggestion.

File: build.gradle (hunk 1 above)
Comment (diguage) [reply build.gradle]: Sorry. I update it.

**Comment by diguage** [comment]
I used the following commands to find the duplicate version declaration.

```shell
grep -E "\(\".*:.*:.*\"\)" -rl . \
  | grep gradle \
  | xargs grep -E "\(\".*:.*:.*\"\)"  {} \
  | awk -F\" '{print q$2}' \
  | sort \
  | uniq -c \
  | sort -nr
```

File: build.gradle [hunk 2]
```diff
@@ -28,24 +28,66 @@ ext {
...
+	aaltoXmlVersion           = "1.1.1"
+	activationApiVersion      = "1.2.0"
+	activationVersion         = "1.2.0"
+	annotationApiVersion      = "1.3.2"
+	aspectjVersion            = "1.9.2"
+	cacheApiVersion           = "1.1.0"
+	commonsFileuploadVersion  = "1.3.3"
+	commonsPoolVersion        = "2.6.0"
+	ejbApiVersion             = "3.2"
+	elApiVersion              = "3.0.1-b04"
```
Comment (diguage) [inline build.gradle]: Which should be bofore? `elApiVersion`? `ejbApiVersion`?

_BTW: I sorted the lines with vim `sort`._

File: build.gradle [hunk 3]
```diff
@@ -28,24 +28,66 @@ ext {
...
+	tiles3Version             = "3.0.8"
+	tomcatVersion             = "9.0.14"
+	transactionApiVersion     = "1.3"
+	underscorejsVersion       = "1.8.3"
+	undertowVersion           = "2.0.16.Final"
+	validationApiVersion      = "1.1.0.Final"
+	websocketApiVersion       = "1.1"
+	woodstoxCoreVersion       = "5.2.0"
+	xmlunitMatchersVersion    = "2.6.2"
+	xstreamVersion            = "1.4.10"
```
Comment (diguage) [inline build.gradle]: _BTW: I used vim `sort` to sort these lines._

**Comment by diguage** [comment]
**I uploaded a picture for comparison.**

image

**Comment by diguage** [comment]
@snicoll  
What's wrong with the PR? Please add your suggestion. Thank you.

**Comment by snicoll** [comment]
@diguage please be patient, it's normal for a PR to stay open for a while so that the team at large can have a look to it before considering merging it.

Reviewing the changes you've pushed since my last review is on my todo list.

**Comment by diguage** [comment]
I use the following command to review the dependences:

```
find . -name "*.gradle" -type f \
  | xargs grep -E "\(\".*:.*:.*\"\)"  {} \
  | awk -F\" '{print $2}' \
  | sort \
  | uniq -c \
  | sort -nr \
  | awk '{printf "%-80s%s\n",$2,$1}' \
  | sort
```

I found：

```
Dependence                                                                      Count
----------------------------------------------------------------------------------------
com.rometools:rome:1.11.1                                                       2
com.rometools:rome:1.12.0                                                       1

org.hibernate:hibernate-core:5.1.16.Final                                       1
org.hibernate:hibernate-core:5.3.7.Final                                        2

org.hibernate:hibernate-validator:5.4.2.Final                                   1
org.hibernate:hibernate-validator:6.0.14.Final                                  4

org.webjars:webjars-locator-core:0.35                                           1
org.webjars:webjars-locator-core:0.36                                           1
```

The dependences have two different version. 

I kept the minority the same as they once was. I unify the majority.

**Shall we unify the dependences to one version?**

**Comment by jhoeller** [comment]
The Rome and WebJars Locator mismatch was an accident; I'll fix those in 5.1.x ASAP.

As for Hibernate Validator, we use distinct versions for being able to retain Bean Validation 1.1 compatibility in `spring-context` itself, just bringing in the Bean Validation 2.0 API through Hibernate Validator 6 in a separate module for integration testing purposes. We'll have to keep differentiating between Hibernate Validator 5 and 6 there, possibly through distinct dependency version constants.

Generally speaking, we tried to avoid an overload of dependency version variables upfront, with only a selection of commonly used ones factored out into that constant section. This hasn't been very consistent, with quite a few versions being redeclared in two modules, so it's definitely worth a revision.

**Comment by diguage** [comment]
@jhoeller 
Could I fix the Rome and WebJars Locator mismatch in the PR?

**Comment by rstoyanchev** [comment]
I believe this is now superseded by https://github.com/spring-projects/spring-framework/commit/d4089747b8e1f4d79497916286c581b8eb69d67f.

**Pull request closed without merging**